### PR TITLE
Feature/options callback

### LINKED ISSUE #15: [Feature Request] Passing a closure to options()
Hey once again!

It would be great if we could pass a closure into the `options()` method.

On my `fields()`I wanted to set a variable using the current class property, because we are not able to use `$this` inside the `options()` since it refers to the actual resource model.

I thought I could tackle this one and if it works as expected I will submit a PR if you are intersted on it.

### PR DESCRIPTION
**Pull request by skoyah**
With this PR users will have the capability of passing a closure to do some logic before outputting the final array.

Resolves #15

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -80,6 +80,21 @@ InlineSelect::make('Status')->options($options)
     ->displayUsingLabels(),
 ```
 
+### Using closures as  `options()` argument
+
+Passing a closure to this method allows data to be formatting based on some specific logic.
```
Comment (brandonferens) [inline readme.md]: Change this sentence to read:
"You may pass a closure to the options method. It must return a key value pair array."

**Review by brandonferens — CHANGES_REQUESTED** [review]

**Comment by brandonferens** [comment]
@skoyah I can't express how glad I am you included a test. Thank you so much!

**Review by brandonferens — APPROVED** [review]

**Comment by skoyah** [comment]
> @skoyah I can't express how glad I am you included a test. Thank you so much!

I love testing. Since I've started learning/writing tests it made my life so much easier!

It also demonstrates I've read the contributing guide! :)

**Comment by skoyah** [comment]
Thanks @brandonferens! 
I'll keep submitting some PR's If find they will bring value to this package!

**Comment by skoyah** [comment]
By the way @brandonferens  could you tag the package to 1.0.6? Did a composer update but the changes are not getting downloaded since is only on dev version.

**Pull request merged**